### PR TITLE
Allow `CliInvoker` redirect build log to file

### PR DESCRIPTION
**Pull request by blindpirate**
In https://github.com/gradle/gradle-profiler/pull/244/files we only added `buildLog` support for `ScenarioInvoker.measureCommandLineExecution`, but not for `CliInvoker`. This PR adds `buildLog` support to it.

### CODE REVIEW

File: src/main/java/org/gradle/profiler/CliInvoker.java [hunk 1]
```diff
@@ -49,15 +55,21 @@ public void runTasks(List<String> tasks, List<String> gradleArgs, List<String> j
 
         ProcessBuilder builder = new ProcessBuilder(commandLine);
         builder.directory(projectDir);
-        if (!daemon) {
+        if (daemon) {
+            builder.environment().put("GRADLE_OPTS", "-Xmx128m -Xms128m -XX:+HeapDumpOnOutOfMemoryError");
```
Comment (wolfs) [inline src/main/java/org/gradle/profiler/CliInvoker.java:59]: Why add these `GRADLE_OPTS` now?

**Review by wolfs — COMMENTED** [review]
I added a question.

File: src/main/java/org/gradle/profiler/CliInvoker.java (hunk 1 above)
Comment (blindpirate) [reply src/main/java/org/gradle/profiler/CliInvoker.java:59]: Same as here: https://github.com/gradle/gradle/commit/f3c5ee6185c19b1e2af44fd8ee90b1c9c8338d26

When the build generates too much log, the log events might congest in client JVM and OOM. gradle-profiler doesn't provide an API to configure the client JVM or environment variables, right?

Comment (wolfs) [reply src/main/java/org/gradle/profiler/CliInvoker.java:59]: Do we already know we need this? If we don't, I wouldn't add it now.

If we need something like that, I would make it configurable via something like `clientJvmArgs` on `CliInvoker`.

Comment (blindpirate) [reply src/main/java/org/gradle/profiler/CliInvoker.java:59]: Yes, we need it for build scan performance test. This is just a workaround, I opened a new issue https://github.com/gradle/gradle-profiler/issues/260 to handle this issue and no-daemon jfr recording issue.

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**